tests/SimplifyTest.elm

```diff
@@ -19373,6 +19374,208 @@ a = Task.succeed << f
+            \() ->
+                """module A exposing (..)
+import Task
+a = Task.map3 f (Task.succeed a) (Task.succeed b) (Task.succeed c)
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "Task.map3 where each task is a succeeding task will result in Task.succeed on the values inside"
+                            , details = [ "You can replace this call by Task.succeed with the function applied to the values inside each succeeding task." ]
+                            , under = "Task.map3"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Task
+a = Task.succeed (f a b c)
+"""
+                        ]
+        , test "should replace c |> g |> Task.succeed |> Task.map3 f (Task.succeed a) (Task.succeed b) by (c |> g) |> f a b |> Task.succeed" <|
+            \() ->
+                """module A exposing (..)
+import Task
+a = c |> g |> Task.succeed |> Task.map3 f (Task.succeed a) (Task.succeed b)
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "Task.map3 where each task is a succeeding task will result in Task.succeed on the values inside"
+                            , details = [ "You can replace this call by Task.succeed with the function applied to the values inside each succeeding task." ]
+                            , under = "Task.map3"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Task
+a = (c |> g) |> f a b |> Task.succeed
+"""
+                        ]
+        , test "should replace Task.map3 f (Task.succeed a) (Task.fail x) task2 by (Task.fail x)" <|
+            \() ->
+                """module A exposing (..)
+import Task
+a = Task.map3 f (Task.succeed a) (Task.fail x) task2
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "Task.map3 where we know the first failing task will result in that failing task"
+                            , details = [ "You can replace this call by the first failing task." ]
+                            , under = "Task.map3"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Task
+a = (Task.fail x)
+"""
+                        ]
+        , test "should replace Task.map3 f (Task.succeed a) (Task.fail x) by always (Task.fail x)" <|
+            \() ->
+                """module A exposing (..)
+import Task
+a = Task.map3 f (Task.succeed a) (Task.fail x)
```

Comment:
We should probably stop using `a` in the references, seeing how that conflicts with the `a =` declaration (making this code that would not compile). Let's try to avoid this in the future and maybe fix this over time, but not necessary for this PR.

Reply:
right, there are so many places with this issue.